patchwork-events-entity/src/main/java/com/patchworkmc/mixin/event/entity/MixinLivingEntity.java

```diff
@@ -74,5 +74,28 @@ private void hookApplyDamageForHurtEventCancel(DamageSource source, float damage
 		if (damage <= 0) {
 			info.cancel();
 		}
-	}
+    }
+
+    @ModifyVariable(method = "handleFallDamage", at = @At(value = "INVOKE", target = "net/minecraft/entity/Entity.handleFallDamage(FF)V", shift = At.Shift.BEFORE), ordinal = 0)
+    private float hookHandleFallDamageDistance(float distance, float damageMultiplier) {
+        LivingEntity entity = (LivingEntity) (Object) this;
+
+        return EntityEvents.onLivingFall(entity, distance, damageMultiplier)[0];
+    }
+
+    @ModifyVariable(method = "handleFallDamage", at = @At(value = "INVOKE", target = "net/minecraft/entity/Entity.handleFallDamage(FF)V", shift = At.Shift.AFTER), ordinal = 1)
+    private float hookHandleFallDamageMultiplier(float distance, float damageMultiplier) {
+        LivingEntity entity = (LivingEntity) (Object) this;
+
+        return EntityEvents.onLivingFall(entity, distance, damageMultiplier)[1];
+    } 
+
+    @Inject(method = "handleFallDamage", at = @At("HEAD"), cancellable = true)
+    private void hookHandleFallDamageCancel(float distance, float damageMultiplier, CallbackInfo info) {
```

Comment:
We should probably only fire this event once. Instead, we could store the values into fields annotated with Unique, and then access them from the ModifyVariable locations.